### ISSUE TITLE
Rule proposal: `no-continuous-array-push`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#937]
Forbid calling `push()` multiple times on the same array.

## Fail

```js
foo.push(1);
foo.push(2);
```

## Pass

```js
foo.push(1, 2);
```

```js
foo.push(1);
length = foo.push(2);
foo.push(3);
```

----
I think it should not work for `.unshift()`, since

```js
foo.unshift(1);
foo.unshift(2);
```

Not equal to
```js
foo.unshift(1, 2);
```

**Comment by sindresorhus** [comment]
> Not equal to

It is if you reverse the argument order.

**Comment by fisker** [comment]
> It is if you reverse the argument order.

Right, I was thinking

```js
foo.unshift(...[1,2]);
foo.unshift(...[3,4]);
```

Not equal to

```js
foo.unshift(...[3,4], ...[1,2]);
```

I was wrong

**Comment by fisker** [comment]
But

``js
foo.unshift(a());
foo.unshift(b());
```

``js
foo.unshift(b(), a());
```

Might not right

**Comment by sindresorhus** [comment]
> foo.unshift(b(), a());

Do you mean if `b()` mutates some global state? Yeah, we can skip function calls.

**Comment by fisker** [comment]
Push may also not safe

```js
foo.push(1);
foo.push(a());
```

If a use `foo`

**Comment by sindresorhus** [comment]
We should still report in these "unsafe" cases as they're very unlikely to happen in real code, but we should use a suggestion instead of auto-fix.

**Comment by sindresorhus** [comment]
This is now accepted.

**Comment by fisker** [comment]
Name?

**Comment by fisker** [comment]
How about let's forbid `.pop` too :smile:

Fail

```js
foo.pop();
foo.pop();
```

Pass

```js
foo.length -= 2;
```

**Comment by futpib** [comment]
I prefer one push call per push as you can easily count pushes at a glance if written this way.

**Comment by fisker** [comment]
Which one you prefer

```js
foo.push(...a);
foo.push(...b);
```

vs

```js
foo.push(...a, ...b);
```

**Comment by sindresorhus** [comment]
@futpib 

```js
foo.push(
    1,
    2
);
```

?

**Comment by silverwind** [comment]
Also similar `concat` usage could warn, thought it's likely even rarer in real code:

```js
foo.concat(bar).concat(baz);
```
```js
let x = foo.concat(bar);
x = x.concat(baz);
```

**Comment by fisker** [comment]
> Also similar concat usage could warn

We are trying to forbid use `.concat` #737, let's ignore it.

**Comment by fisker** [comment]
I'll open a new issue about `.shift()` to see if some one need it.

**Comment by fisker** [comment]
#1019
